examples/matmul.py

```diff
@@ -20,17 +32,60 @@ def matmul(x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
         acc = hl.zeros([tile_m, tile_n], dtype=torch.float32)
         for tile_k in hl.tile(k):
             acc = torch.addmm(acc, x[tile_m, tile_k], y[tile_k, tile_n])
-        out[tile_m, tile_n] = acc
+        out[tile_m, tile_n] = epilogue(acc, [tile_m, tile_n])
     return out
 
 
+# For unit test compatibility
+matmul = matmul_with_epilogue
+
+
+def autotune(m: int, k: int, n: int) -> None:
+    x = torch.randn([m, k], device="cuda", dtype=torch.float16)
+    y = torch.randn([k, n], device="cuda", dtype=torch.float16)
+    bias = torch.randn([n], device="cuda", dtype=torch.float16)
+    args = (x, y, lambda acc, tile: torch.relu(acc + bias[tile[1]]))
+    best_config = matmul_with_epilogue.autotune(args, force=True)
+    print(f"Best config: {best_config}")
+    best_config.save("best_config.json")
+
+
 def check(m: int, k: int, n: int) -> None:
     x = torch.randn([m, k], device="cuda", dtype=torch.float16)
     y = torch.randn([k, n], device="cuda", dtype=torch.float16)
-    run_example(matmul, torch.matmul, (x, y))
+    bias: torch.Tensor = torch.randn([n], device="cuda", dtype=torch.float16)
+
+    # Test without bias
+    kernel_no_bias = lambda x, y: matmul(x, y)  # noqa: E731
+    expected_no_bias = lambda x, y: torch.matmul(x, y)  # noqa: E731
+    run_example(kernel_no_bias, expected_no_bias, (x, y))
+
+    # Test with bias
+    bias = torch.randn([n], device="cuda", dtype=torch.float16)
+    kernel_with_bias = lambda x, y: matmul(x, y, lambda acc, tile: acc + bias[tile[1]])  # noqa: E731
```

Comment:
I feel that having `bias: [n]` and do `bias[tile[1]]` here is more common, but if we explicitly want to demonstrate that `bias[tile]` where tile is a list (`[tile_m, tile_n]`) can work, I can change it back too (it's also already tested in test_examples.py `test_template_via_closure0`).

Reply:
If you pass a [1, N] bias tensor it will broadcast.